components/Input/Input.jsx

```diff
@@ -89,11 +89,28 @@ class Input extends React.Component {
   constructor(props) {
     super(props);
 
-    const { type, id } = props;
+    const { type, id, value } = props;
 
-    this.state = { type };
+    this.state = {
+      type,
+      isDirty: !!value,
```

Comment:
`hasValue` would be easier to understand

Reply:
Is this state really necessary? Maybe you can validate this just using `currentValue`